dartagnan/src/main/java/com/dat3m/dartagnan/solver/caat/misc/IteratorPool.java

```diff
@@ -0,0 +1,35 @@
+package com.dat3m.dartagnan.solver.caat.misc;
+
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.function.Function;
+
+//TODO: We probably do not need this anymore
```

Comment:
I need to double check. But IIRC it is still in use. This class came about after profiling `Saturation` and seeing that over 1gb of just iterator objects are created over extended periods of time (which may or may not be problematic). But the new approach (letting the solver generate the co) causes less search and hence also less iteration. I need to perform new profiling to see if this class has any merit anymore but my initial hunch was that it may not be needed anymore (that's why I added the comment)